src/common/components/paymentMethods/creditCardForm/creditCardForm.component.js

```diff
@@ -96,11 +97,10 @@ class CreditCardController {
       this.creditCardPaymentForm.expiryMonth.$validate();
     });
 
-    if(!this.paymentMethod) {
-      this.creditCardPaymentForm.securityCode.$parsers.push(this.paymentValidationService.stripNonDigits);
-      this.creditCardPaymentForm.securityCode.$validators.minlength = number => this.paymentMethod && !this.creditCardPayment.cardNumber || toString(number).length >= 3;
-      this.creditCardPaymentForm.securityCode.$validators.maxlength = number => toString(number).length <= 4;
-    }
+    this.creditCardPaymentForm.securityCode.$parsers.push(this.paymentValidationService.stripNonDigits);
+    this.creditCardPaymentForm.securityCode.$validators.minlength = number => this.paymentMethod && !this.creditCardPayment.cardNumber || toString(number).length >= 3;
```

Comment:
I think you can toss `this.paymentMethod && !this.creditCardPayment.cardNumber || ` as that was only validating this input when the cardNumber was changed.